components/tooltip/README.md

```diff
@@ -0,0 +1,66 @@
+```html
+<script type="module">
+  import '@brightspace-ui/core/components/tooltip/tooltip.js';
+</script>
+
+<div class="offset-parent">
+	<button id="tooltip-boundary">Tooltip boundary</button>
+	<d2l-tooltip for="tooltip-boundary"
+		boundary="{&quot;top&quot;:50, &quot;bottom&quot;:10, &quot;left&quot;:100, &quot;right&quot;:450}">
+		This tooltip will not expand beyond its boundaries unless it is impossible to fit it inside
+	</d2l-tooltip>
+</div>
+```
+
+
+**Properties:**
+* `for` (required, String): provide the `id` of the tooltip's target element. If this attribute is not provided, the tooltip's parent element will be used as its target.
+* `align` (String): optionally align the tooltip with either the start or end of its target. If not set, the dropdown will attempt be centred. Valid values are: `start` and `end`.
+* `boundary` (Object) - optionally provide boundaries to where the tooltip will appear. Valid properties are `"top"`, `"bottom"`, `"left"` and `"right"`. The boundary is relative to the tooltip's [offset parent](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetParent).
+* **DEPRECATED** `custom-target` (Object) - optionally provide a rectangle that the tooltip should position itself around. The object must provide values for `"left"`, `"top"`, `"width"`, and `"height"`. The custom target is relative to the tooltip's [offset parent](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetParent). If a custom target is provided the user is responsible for showing and hiding the tooltip since it no longer has a target element that can be focused or hovered.
+* `delay` (Number, default: `0`) - optionally provide a delay that to prevent the tooltip from opening immediately when hovered. This delay will only apply to hover, not focus.
+* `disable-focus-lock` (Boolean, default: `false`) - disables focus lock so that the tooltip will automatically close when no longer hovered even if it still has focus.
+* `force-show` (Boolean, default: `false`): force the tooltip to stay open as long as it remains `true`
+* `offset`: (Number, default: `16.5`): adjust the size of the gap between the tooltip and its target.
+* `position` (String): optionally force the tooltip to open in a certain direction. Valid values are: `top`, `bottom`, `left` and `right`. If no position is provided, the tooltip will open in the first position that has enough space for it to open in the order: top, bottom, right, left.
+* `showing` (Boolean, default: `false`): a **readonly** only property that can be used to determine if the tooltip is currently visible
+* `state` (String, default: `info`): the style of the tooltip based on the type of information it displays. Valid values are: `info` and `error`.
+
+**Events:**
+* `d2l-tooltip-show`: dispatched when the tooltip is opened
+* `d2l-tooltip-hide`: dispatched when the tooltip is closed
+
+**CSS Variables:**
```

Comment:
I'm good with not even mentioning things we don't want people to use.

Reply:
The reasoning here was also the same reason for documenting `custom-target`. I'm hoping, possibly naively, if people see them they'll understand why we don't want people to use them and that their is a proper route to solve their problem by opening an issue and adding new `state` values.

Reply:
I'm going to remove the documentation for these and move the quote to the `state` property.

> If you find yourself needing a style that isn't supported by the `state` attribute please create a Github issue.